.github/workflows/large-files.yml

```
@@ -0,0 +1,39 @@
+name: Check for large files
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+jobs:
+  check_large_files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

```suggestion
      - name: Checkout repository
        uses: actions/checkout@v3
        with:
            fetch-depth: 0
```

I think this is needed to solve the `fatal: origin/main...HEAD: no merge base` issue.
However, it may make the test take long to run.

I'm thinking that it would be nice to get it working, and we can optimise later.

---

oops, missed that update broke it. it was a permission issue before. hmm if we use the default 2 dot diff I believe it could give errant warnings if the feature branch isn't up to date with main.

That might be preferable to `fetch-depth: 0` though if its rare enough?